catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/MultilingualTest.java

```diff
@@ -0,0 +1,150 @@
+		String buttonContinueName = getResourceString(R.string.main_menu_continue);
+		String buttonNewName = getResourceString(R.string.main_menu_new);
+		onView(withId(R.id.main_menu_button_continue)).check(matches(withText(containsString(buttonContinueName))));
+		onView(withId(R.id.main_menu_button_new)).check(matches(withText(buttonNewName)));
+		onView(withId(R.id.main_menu_button_programs)).check(matches(withText(buttonProgramsName)));
+	}
+
+	@Test
+	public void testChangeLanguageToUrdu() {
+		gotoMultilingualActivity();
+		onData(anything()).inAdapterView(withId(R.id.list_languages)).atPosition(45).perform(click());
+		assertThat("current Locale is not Urdu", Locale.getDefault().getDisplayLanguage(), is(URDULOCALE.getDisplayLanguage()));
+		assertThat("not RTL", true, is(isRTL()));
+		SystemClock.sleep(shortSleep);
+		String buttonProgramsName = getResourceString(R.string.main_menu_programs);
+		String buttonContinueName = getResourceString(R.string.main_menu_continue);
+		String buttonNewName = getResourceString(R.string.main_menu_new);
+		onView(withId(R.id.main_menu_button_continue)).check(matches(withText(containsString(buttonContinueName))));
+		onView(withId(R.id.main_menu_button_new)).check(matches(withText(buttonNewName)));
+		onView(withId(R.id.main_menu_button_programs)).check(matches(withText(buttonProgramsName)));
+	}
+
+	@Test
+	public void testChangeLanguageToFarsi() {
+		gotoMultilingualActivity();
+		onData(anything()).inAdapterView(withId(R.id.list_languages)).atPosition(46).perform(click());
+		assertThat("current Locale is not Farsi", Locale.getDefault().getDisplayLanguage(), is(FARSILOCALE.getDisplayLanguage()));
+		assertThat("not RTL", true, is(isRTL()));
+		SystemClock.sleep(shortSleep);
+		String buttonProgramsName = getResourceString(R.string.main_menu_programs);
+		String buttonContinueName = getResourceString(R.string.main_menu_continue);
+		String buttonNewName = getResourceString(R.string.main_menu_new);
+		onView(withId(R.id.main_menu_button_continue)).check(matches(withText(containsString(buttonContinueName))));
+		onView(withId(R.id.main_menu_button_new)).check(matches(withText(buttonNewName)));
+		onView(withId(R.id.main_menu_button_programs)).check(matches(withText(buttonProgramsName)));
+	}
+
+	private static boolean isRTL() {
+		return isRTL(Locale.getDefault());
+	}
+
+	private static boolean isRTL(Locale locale) {
+		final int directionality = Character.getDirectionality(locale.getDisplayName().charAt(0));
+		return directionality == Character.DIRECTIONALITY_RIGHT_TO_LEFT
+				||
+				directionality == Character.DIRECTIONALITY_RIGHT_TO_LEFT_ARABIC;
+	}
+
+	private String getResourceString(int id) {
+		Context targetContext = InstrumentationRegistry.getTargetContext();
+		return targetContext.getResources().getString(id);
+	}
+
+	private void gotoMultilingualActivity() {
+		openActionBarOverflowOrOptionsMenu(getInstrumentation().getTargetContext());
```

Comment:
please have a look at SettingsActivityTest on how to directly open the settings activity, and eg. DeleteSoundDialogTest for on how to directly open a certain fragment for a test.